Terasology/Alchemy
# Drink potion system updates + UI improvements + Documentation

### Contains

* InGameHelp and WorkstationInGameHelp added to all items.
* Moved base definition of potion containers over to Potions. The ones here are deltaed.
* Added HerbalismStationWindowSystem and a method in the associated HerbalismStationWindow to speed up responsiveness of the HerbalismStation.
* When brewing potions, the original potion bottle's prefab name and durability are now stored in the resultant potion's PotionComponent.
* Potion recipe prefabs have now been updated to require the inclusion of a potion bottle.
* Documentation has been added to all classes.

### Note

For those testing this, **make sure** to have the updates in [Potions PR #3](https://github.com/Terasology/Potions/pull/3) downloaded.

Also, prepend @API to the InteractionScreenComponent.java file. Otherwise, you may end up with the following error:

> [main] ERROR o.t.module.sandbox.ModuleClassLoader - Denied access to class (not allowed with this module's permissions): org.terasology.logic.characters.interactions.InteractionScreenComponent

### How to Test

**Testing the new potions system:**
* Spawn a PortableHerbalismStation using the console. 
* Then, spawn a WoodenBucket and some sticks. 
* Fill up the WoodenBucket at the nearest water source, and add the water to the station. 
* Spawn some "HealingHerb"s and "GlassBottle"s. 
* Add the sticks to the fuel source, and the rest of the items to the input. 
* Create some HealPotions and drink them. No bottle should be returned.
* Spawn some "ToughGlassBottle"s and replace the "GlassBottles" with them.
* Create some HealPotions using the ToughGlassBottles.
* Drink those potions. For each potion drunk, a ToughGlassBottle with a durability of 6/9 should be returned.

**Misc:**
* Proofread the documentation.

### Issues

None so far.

### Credits

@AWebb3701 and @Cervator for advice.
